### PR TITLE
Add base-url config option to run server on subpath

### PR DESCRIPTION
**Pull request by willwest**
Addresses issue #48

### CODE REVIEW

**Comment by willwest** [comment]
@rickbergfalk Closing this for now, will reopen when it's ready for review.

File: README.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ By default SqlPad will listen from all available addresses (0.0.0.0). This may b
 
 By default SqlPad will use port 80. This may be overridden via cli parameter `--port` or environment variable `SQLPAD_PORT`.
 
+### Base URL Path
+
+By default SqlPad will be accessible from the absolute base path of the available IP address(es). This may be overridden via cli parameter `--base` or environment variable `SQLPAD_BASE_URL`.
```
Comment (ybroze) [inline README.md]: I'd recommend using `--base-url` if the envvar is called `SQLPAD_BASE_URL`.

File: server.js [hunk 2]
```diff
@@ -166,20 +170,22 @@ app.use('/config', mustBeAdmin);
...
-require('./routes/onboarding.js')(app);
-require('./routes/user-admin.js')(app);
-require('./routes/connections.js')(app);
-require('./routes/queries.js')(app);
-require('./routes/run-query.js')(app); // ajaxy route used for executing query and getting results
-require('./routes/download-results.js')(app); // streams cached query results to browser
-require('./routes/schema-info.js')(app);
-require('./routes/configs.js')(app);
-require('./routes/tags.js')(app);
+require('./routes/oauth.js')(router, passport, router);
```
Comment (ybroze) [inline server.js]: Does one of these need to be the `app`?

**Comment by rickbergfalk** [comment]
Thanks @willwest! I'll get this merged and published sometime tomorrow. 

I agree with supporting the cli flag --base-url to align with the matching env var. Feel free to make the change if you'd like otherwise I can take care of it after the merge. 

Thanks again!

**Comment by willwest** [comment]
@rickbergfalk Cool, thanks! I'll make that change and do a little more testing before we merge. Since there's no automated testing (yet), and this change touches almost every part of the code, I want to confirm that this doesn't introduce any new bugs.

**Comment by rickbergfalk** [comment]
@willwest Perfect! I can help test as well - I'll plan on running the changes on my local SqlPad instance prior to publish. 

Automated testing is something that's on my todo list for the summer. If you have any ideas, experience, or preferences on that let me know.

**Comment by rickbergfalk** [comment]
@willwest did some testing and discovered the user signup/signin stuff wasn't fully considering baseUrl. 

I think I got it mostly working. Sadly my git skills aren't that strong. Is there a way I can amend your pull request with the changes? Or are you able to pull them in from the branch I (well the GitHub app) created? 

see this commit for details:
https://github.com/rickbergfalk/sqlpad/commit/8661deb6360763c17a832237ca4006abe2a59c99

**Comment by rickbergfalk** [comment]
@willwest Any updates on this? 

If everyone's alright with the approach, I can merge this in and apply a couple fixes post merge. If anyone has a more preferable way of managing this let me know :)

**Comment by willwest** [comment]
@rickbergfalk Sorry I've been absent on this. Still want to take a closer look before merging, but expect a comment / maybe some updates this week. Hopefully will have more time to work on this repo gradually over the next few weeks.

**Comment by willwest** [comment]
@rickbergfalk Are the changes in 8661deb6360763c17a832237ca4006abe2a59c99 the only ones needed to make this work? I'll cherrypick the commit onto this PR, and we can merge them in together. Wouldn't be good to have an unstable build in master, even if it's for a short amount of time.

**Comment by rickbergfalk** [comment]
@willwest I believe so. The only case I didn't test was the google OAuth authentication.

File: client-js/connection.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ module.exports = function () {
         renderTesting();
         $.ajax({
             type: "POST",
-            url: "/connections/test",
+            url: baseUrl + "/sqlpad/connections/test",
```
Comment (rickbergfalk) [inline client-js/connection.js]: missed this with my initial testing. looks like /sqlpad snuck in during initial development

Comment (willwest) [inline client-js/connection.js]: Thanks, fixed! Once we're comfortable with OAuth authentication, this should be good to go.

**Comment by rickbergfalk** [comment]
I'll be taking another look this weekend and will try to test out OAuth. Hope to merge by monday. Thanks again @willwest for diving into this one!

**Comment by rickbergfalk** [comment]
Finally got around to testing OAuth - see https://github.com/willwest/sqlpad/pull/1

**Comment by rickbergfalk** [comment]
Woo! Thanks @willwest!

**Pull request merged**